Zig.gitignore

```diff
@@ -0,0 +1,4 @@
+# This file is for zig-specific build artifacts.
+
+zig-cache/
+zig-out/
```

Comment:
`zig-*` would be better

Reply:
I think explicitness, in this case, is better than being implicit. I can see a scenario where someone would include a `zig-something` directory to use as a package or maybe create a `zig-build` directory to hold helpers for `build.zig`.

Reply:
Or could be `zig-{cache,out}` 🤔🤔

Reply:
It's fine as it is...

Reply:
> I can see a scenario where someone would include a `zig-something` directory to use as a package

should be in a `libs` or `deps` folder. additionally people should not be venturing to begin with.

> or maybe create a `zig-build` directory to hold helpers for `build.zig`.

do you have even one example of this? feels very contrived

Reply:
> should be in a `libs` or `deps` folder
Why should something as simple as a gitignore constrain project structure?

> additionally people should not be vendoring to begin with.
While I would not vendor, vendoring is a completely valid thing to do. And having the default gitignore shoot people in the foot is bad.

Reply:
> do you have even one example of this? feels very contrived

No I don't have any examples - I'm not sure such an example exists. Yes, both my points are contrived. Nonetheless it's not technically difficult to end up with either of my examples, nor super far fetched. Edge cases matter. And in this case, edge cases are covered by being explicing, ignoring both paths. 

I'm not trying to prove you wrong, nektro, I just don't think the few saved bytes does anything good should someone deviate (greatly) from "how things are done".

Reply:
Agreeing with Lee here. The gitignore shouldn't dictate project structure. It should ignore files generated by the compiler, and that is all it should do. We shouldn't lock people out of using submodules or vendoring their dependencies.